### DISCUSSION TITLE
멀티플레이 영상 가이드

### DISCUSSION THREAD

**Discussion by Zepeto-TechEvangelist** [naverz/zepeto-studio-kor#459] [2. Official Samples]
안녕하세요~ 제페토 크리에이터 여러분 😄
멀티플레이 월드 제작에 도움을 드리기 위해 차근차근 따라하실 수 있는 영상 가이드를 준비했습니다! 
[Multiplay](https://studio.zepeto.me/kr/guides/multiplay) 가이드 문서와 함께 보시면 더 많은 도움이 되실 것입니다.

### [ 1편 : 멀티플레이 설정하기 ]
[![🎥 영상 보기](https://user-images.githubusercontent.com/100118650/156959385-c756ec6c-6e40-467e-b61a-e21468a971ac.png)](https://youtu.be/S68B1TMa-A8)

- 🎥 영상 보기 : https://youtu.be/S68B1TMa-A8
- 멀티플레이 서버 생성부터 클라이언트 생성까지, 멀티플레이 월드 개발을 위한 환경 설정을 진행합니다.

<br></br>

### [ 2편 : 월드 로직 작성하기 1 ]
[![🎥 영상 보기](https://user-images.githubusercontent.com/100118650/156959395-538e12b7-43c9-43d5-9b0f-9cf4cfc7304a.png)](https://youtu.be/tfkY_raboV0)
- 🎥 영상 보기 : https://youtu.be/tfkY_raboV0
- 서버, 클라이언트 간 통신을 위해 필요한 Schema에 대해 알아보고, Schema Types와 Room State를 정의합니다.

1. Room Lifecycle을 통해 서버에서 기본적인 이벤트 처리 코드를 작성합니다.
2. 또한 서버에 데이터를 저장하는 방법을 알아봅니다.
3. 클라이언트에서 Room 생성 및 입장 시의 이벤트 처리 코드를 작성합니다.
4. 또한 Room State를 통해 서버의 변경사항을 처리하는 코드를 작성합니다.
5. Room message를 통해 클라이언트에서 서버로 State를 전송하는 방법을 알아봅니다.
6. 서버에서 message를 수신받아 State를 관리하는 방법을 알아봅니다.

<br></br>

### [ 3편 : 월드 로직 작성하기 2 ] 
[![🎥 영상 보기](https://user-images.githubusercontent.com/100118650/156959410-a139598e-a270-46cf-95fd-2267b2c544cd.png)](https://youtu.be/H92Gd2G9DhM)
- 🎥 영상 보기 : https://youtu.be/H92Gd2G9DhM
- 플레이어의 위치 동기화부터 플레이어의 퇴장까지를 진행합니다.

1.  클라이언트의 현재 위치를 일정 시간 마다 서버로 전송합니다.
2. 그리고 서버에서 수신 받은 다른 플레이어의 위치를 동기화합니다.
3. 마지막으로 플레이어의 퇴장에 대한 처리도 진행해 월드 로직을 마무리합니다.

<br></br>

### [ 4편 : 서버 구동 및 멀티플레이 접속하기 ] 
[![🎥 영상 보기](https://user-images.githubusercontent.com/100118650/156959421-a98e9532-2abd-4ebd-9ff3-896d60cf5aac.png)](https://youtu.be/FmK6JNlSjJA)
-  🎥 영상 보기 : https://youtu.be/FmK6JNlSjJA
- 이번 편에서는 서버를 구동하고 접속하여 멀티플레이를 진행합니다.

1. 먼저 서버를 구동하고 접속하는 방법을 알아봅니다.
2. 그리고 QR코드 접속 혹은 IP주소로 접속 할 수 있는 custom environment 방식으로 멀티플레이 환경에 접속해봅니다.

<br></br>

### [ 5편 : 멀티플레이 심화:IWP 서버 연동과 서버 DataStorage ]
[![🎥 영상 보기](https://user-images.githubusercontent.com/100118650/156959427-37d9aaa5-d631-40de-90d9-3754b59ade4c.png)](https://youtu.be/d-MaqjP-vjA)
-  🎥 영상 보기 : https://youtu.be/d-MaqjP-vjA
- 멀티플레이에서 사용 가능한 IWP와 DataStorage를 활용해봅니다.

1. 월드 상품을 구매할 수 있는 IWP 기능을 통해 월드에 물약을 만들고, 구매 시 플레이어 이동 속도가 증가하도록 구현해 봅니다.
2. 플레이어의 데이터를 서버에 저장해주는 DataStorage를 통해 플레이어의 이동 속도를 저장하고 이후 입장 시에 데이터를 불러오는 과정을 진행해 봅니다.

**Comment by rjc90** [comment]
감사합니다.!!!

**Comment by hipsc159** [comment]
애니메이션을 따로 만들었을때, 움직임이 다른 사람들과 공유하려면 어떻게 해야하나요?

**Comment by chlguswls32** [comment]
감사합니다!!
